lib/index.js

```diff
@@ -201,6 +201,53 @@ function getOptions(options, cb) {
 }
 
 /**
+ * Returns function to be called when executing a custom importer
+ *
+ * @param {Object} target
+ * @param {Object} options
+ * @api private
+ */
+
+function establishImporter(target, options) {
+  return function(file, prev, bridge) {
+    function done(data) {
+      bridge.success(data);
+    }
+
+    var result = target.call(options.context, file, prev, done);
+
+    if (result) {
+      done(result === module.exports.NULL ? null : result);
+    }
+  };
+}
+
+/**
+ * Returns function to be called when synchronously executing a custom importer
+ *
+ * @param {Object} target
+ * @param {Object} options
+ * @api private
+ */
+
+function establishImporterSync(target, options) {
+  return function(file, prev) {
+    var resultCb;
+    function done(data) {
+      resultCb = data;
+    }
+
+    var result = target.call(options.context, file, prev, done);
+    result = (typeof resultCb !== 'undefined') ? resultCb : result;
+
+    if(typeof result !== 'undefined') {
+      return result === module.exports.NULL ? null : result;
+    }
+    return new Error('no value returned by importer: possibly due to async operation');
```

Comment:
I'm not a fan of this error message. Given the use case for this message to appear there isn't much the user can do about it.

We should provide enough context for the user to action on this error message. At a minimum we should should supply the file that failed to import.